modules/drivers/mongo/src/metabase/driver/mongo.clj

```diff
@@ -219,45 +145,215 @@
                     :value %}))
            set))))
 
-(defn- sample-documents [^MongoDatabase db table sort-direction]
-  (let [coll (mongo.util/collection db (:name table))]
-    (mongo.util/do-find coll {:keywordize true
-                              :limit metadata-queries/nested-field-sample-limit
-                              :skip 0
-                              :sort-criteria [[:_id sort-direction]]
-                              :batch-size 256})))
-
-(defn- table-sample-column-info
-  "Sample the rows (i.e., documents) in `table` and return a map of information about the column keys we found in that
-   sample. The results will look something like:
-
-      {:_id      {:count 200, :len nil, :types {java.lang.Long 200}, :semantic-types nil, :nested-fields nil},
-       :severity {:count 200, :len nil, :types {java.lang.Long 200}, :semantic-types nil, :nested-fields nil}}"
-  [^MongoDatabase db table]
-  (try
-    (reduce
-     (fn [field-defs row]
-       (loop [[k & more-keys] (keys row), fields field-defs]
-         (if-not k
-           fields
-           (recur more-keys (update fields k (partial update-field-attrs (k row)))))))
-     (ordered-map/ordered-map)
-     (concat (sample-documents db table 1) (sample-documents db table -1)))
-    (catch Throwable t
-      (log/error (format "Error introspecting collection: %s" (:name table)) t))))
+(defn- sample-stages
+  "Query stages which get a sample of the data in the collection, of size `n`. Half of the sample is from the first
+   inserted documents and the other half from the last inserted documents."
```

Comment:
Nitpick https://guide.clojure.style/#docstring-indentation

Only pointing this out because enabling `cljfmt` against our codebase is something I'll probably work on in the near future as part of DevEx stuff and I don't want to have any more needless churn than we need